Mathlib/RingTheory/Ideal/KrullDimension.lean

```diff
@@ -0,0 +1,100 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fangming Li, Jujian Zhang
+-/
+import Mathlib.AlgebraicGeometry.PrimeSpectrum.Basic
+import Mathlib.Topology.KrullDimension
+
+/-!
+# Krull dimensions of (commutative) rings
+
+Given a commutative ring, its ring theoretic Krull dimension is the order theoretic Krull dimension
+applied to its prime spectrum. Unfolding this definition, it is the length of the longest series of
+prime ideals ordered by inclusion.
+
+Key results in this file include:
+1. the ring theoretic Krull dimension of a commutative ring is equal to the topological Krull
+   dimension of its prime spectrum;
+2. given a surjective homomorphism `f : R →+* S` of commutative rings, the Krull dimension of `S`
+   cannot exceed that of `R`;
+3. the Krull dimension of any quotient ring of a commutative ring must be less than or equal to
+   the Krull dimension of that commutative ring;
+4. two isomorphic commutative rings have the same Krull dimension;
+5. for a commutative ring whose Krull dimension is zero, any prime ideal is maximal;
+6. for a commutative ring whose Krull dimension is zero, any prime ideal is minimal;
+7. the Krull dimension of a field is zero.
+-/
+
+/--
+The ring theoretic Krull dimension is the Krull dimension of its spectrum ordered by inclusion.
+-/
+noncomputable abbrev ringKrullDim (R : Type*) [CommRing R] : WithBot (WithTop ℕ) :=
+  krullDim (PrimeSpectrum R)
+
+namespace ringKrullDim
```

Comment:
I am a bit unclear about whether this is the right approach for naming. For example with `finprod` we just have loads of theorems whose names start with `finprod_` rather than a namespace `finprod`.

Reply:
I have noticed that [RingTheory/Multiplicity](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Multiplicity.html#multiplicity) uses the same naming method as mine. So I don't think I need to change the way I name theorems here.